packages/core/src/wallet/IndyWallet.ts

```diff
@@ -96,6 +94,20 @@ export class IndyWallet implements Wallet {
 
     try {
       await this.indy.createWallet({ id: walletConfig.id }, { key: walletConfig.key })
+
+      this.walletConfig = {
+        id: walletConfig.id,
+        key: walletConfig.key,
+      }
+
+      // We usually want to create master secret only once, therefore, we can to do so when creating a wallet.
+      await this.open(walletConfig)
+
+      // We need to open wallet before creating master secret because we need wallet handle here.
+      await this.createMasterSecret(this.handle, walletConfig.id)
+
+      // We opened wallet just to create master secret, we can close it now.
+      await this.close()
```

Comment:
We need to open the wallet to create a master secret. I think it's a good trade-off to do so in here. The create wallet method should be called only once (or at least fewer times) than the open wallet method. It's a question, if we should close the wallet to avoid user's confusion or assume that the developer will want to call `open` operation after `create` and keep it open to save some computer time.